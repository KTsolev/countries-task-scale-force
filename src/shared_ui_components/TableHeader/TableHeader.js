
import React from 'react';

export const TableHeader = (props) => {
    const { onPress, clasName } = props;
    return (
        <div className={clasName}>
            <span 
                className='list-header-item'
                data-prop='Iso Code'
                onClick={onPress}
            >
                Iso Code
            </span>
            <span 
                className='list-header-item'
                data-prop='Name'
                onClick={onPress}
            >
                Name
            </span>    
            <span 
                className='list-header-item'
                data-prop='Capital City'
                onClick={onPress}
            >
             Capital City
            </span>    
            <span 
                className='list-header-item'
                data-prop='Longitude'
                onClick={onPress}
            >
                Longitude
            </span>    
            <span 
                className='list-header-item'
                data-prop='Latitude'
                onClick={onPress}
            >
                Latitude
            </span>
        </div>
    );
}