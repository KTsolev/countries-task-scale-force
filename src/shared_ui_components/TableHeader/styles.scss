@import '../../styles/mixins/mixins.scss';

.table {
    &-header {

        @include row; 
    
        &-item {
    
            @include row-item;
        
            font-size: 18px;
            font-weight: 700;
            background-color: #106c97;
             color: #ffe4c4;
            padding: 4px;
    
            &:hover,
            &:visited {
               @include hover-mixin(#545f6d, #e9e64b);
            }
        }

        &--vertical {
            display: none;
        }
     }
}