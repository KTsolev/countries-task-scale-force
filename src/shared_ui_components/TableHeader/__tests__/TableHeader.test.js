import React from 'react';
import { TableHeader } from '../TableHeader';
import renderer, { create, act } from 'react-test-renderer';

const onPress = jest.fn();

describe('TableHeader tests', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<TableHeader />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when all props are passed', () => {
    const tree = renderer
      .create(<TableHeader onPress={onPress} className='row' />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should handle onclick', () => {
    const tree = create(<TableHeader onPress={onPress} />).root;

    const button = tree.findAllByType('span')[1];

    expect(tree.findAllByType('span').length).toEqual(5);
    
    act(button.props.onClick);

    expect(onPress).toHaveBeenCalled();
  });
})
