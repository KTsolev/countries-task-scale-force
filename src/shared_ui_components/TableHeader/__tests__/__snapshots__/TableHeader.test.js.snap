// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader tests renders correctly 1`] = `
<div>
  <span
    className="list-header-item"
    data-prop="Iso Code"
  >
    Iso Code
  </span>
  <span
    className="list-header-item"
    data-prop="Name"
  >
    Name
  </span>
  <span
    className="list-header-item"
    data-prop="Capital City"
  >
    Capital City
  </span>
  <span
    className="list-header-item"
    data-prop="Longitude"
  >
    Longitude
  </span>
  <span
    className="list-header-item"
    data-prop="Latitude"
  >
    Latitude
  </span>
</div>
`;

exports[`TableHeader tests when all props are passed 1`] = `
<div>
  <span
    className="list-header-item"
    data-prop="Iso Code"
    onClick={[MockFunction]}
  >
    Iso Code
  </span>
  <span
    className="list-header-item"
    data-prop="Name"
    onClick={[MockFunction]}
  >
    Name
  </span>
  <span
    className="list-header-item"
    data-prop="Capital City"
    onClick={[MockFunction]}
  >
    Capital City
  </span>
  <span
    className="list-header-item"
    data-prop="Longitude"
    onClick={[MockFunction]}
  >
    Longitude
  </span>
  <span
    className="list-header-item"
    data-prop="Latitude"
    onClick={[MockFunction]}
  >
    Latitude
  </span>
</div>
`;
